components/GraphDiagram/GraphDiagram.tsx

```diff
@@ -89,14 +89,14 @@ export default function GraphDiagram({ activity }: { activity: LoadActivity }) {
   // Signal for when Graph DOM changes
   const [domSignal, setDomSignal] = useState(0);
 
-  async function handleGraphClick(event: React.MouseEvent) {
-    const target = event.target as HTMLDivElement;
+  async function handleGraphClick(event: React.MouseEvent<HTMLElement>) {
+    const target = event.currentTarget;
```

Comment:
I also remove these assertions and used native React types.

One thing to keep in mind is that `target` is the exact top-most element clicked, while `currentTarget` is the element where the listener was registered. From a quick look, it seems that this is fine.